src/index.ts

```diff
@@ -64,17 +64,12 @@ const defaultOptions: Options = {
 };
 
 function jsQR(data: Uint8ClampedArray, width: number, height: number, providedOptions: Options = {}): QRCode | null {
-
-  const options = defaultOptions;
-  Object.keys(options || {}).forEach(opt => { // Sad implementation of Object.assign since we target es5 not es6
-    (options as any)[opt] = (providedOptions as any)[opt] || (options as any)[opt];
-  });
-
-  const shouldInvert = options.inversionAttempts === "attemptBoth" || options.inversionAttempts === "invertFirst";
-  const tryInvertedFirst = options.inversionAttempts === "onlyInvert" || options.inversionAttempts === "invertFirst";
+  const { inversionAttempts = defaultOptions.inversionAttempts } = providedOptions;
+  const shouldInvert = inversionAttempts === "attemptBoth" || inversionAttempts === "invertFirst";
+  const tryInvertedFirst = inversionAttempts === "onlyInvert" || inversionAttempts === "invertFirst";
   const {binarized, inverted} = binarize(data, width, height, shouldInvert);
   let result = scan(tryInvertedFirst ? inverted : binarized);
-  if (!result && (options.inversionAttempts === "attemptBoth" || options.inversionAttempts === "invertFirst")) {
+  if (!result && (inversionAttempts === "attemptBoth" || inversionAttempts === "invertFirst")) {
```

Comment:
I think this can be simplified to:
```diff
-if (!result && (inversionAttempts === "attemptBoth" || inversionAttempts === "invertFirst")) {
+if (!result && shouldInvert) {
```